extensions/util.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+import pandas as pd
+from mbi import Domain, Dataset
+import itertools
+
+def powerset(iterable):
+    s = list(iterable)
+    return itertools.chain.from_iterable(itertools.combinations(s, r) for r in range(1,len(s)+1))
+
+def downward_closure(cliques):
+    ans = set()
+    for proj in cliques:
+        ans.update(powerset(proj))
+    return list(sorted(ans, key=len))
+
+def discretize(df, schema, clip=None):
+
+    weights = None
+    if clip is not None:
```

Comment:
let's remove clipping functionality I think, it didn't end up helping that much anyway

Reply:
Okey dokey! "clip" has been removed.